### ISSUE TITLE
mixpanel.reset() firefox issue

### ISSUE THREAD

**Issue by eballeste** [mixpanel/mixpanel-js#108]
mixpanel.reset() is called when users logout in order to prevent profile hijacking and polluting the alias table since we've found situations where more than one user could sign in from the same computer/browser.

however, when testing the reset() method in firefox, instead of resetting the distinct_id, a new cookie is created with an additional distinct_id. 

![screen shot 2016-11-14 at 6 50 53 pm](https://cloud.githubusercontent.com/assets/6924108/20287633/95f32c76-aa9b-11e6-94bf-4ddea30ef3d9.png)

This issue only seems to be happening with firefox. Works as intended in safari/chrome. Have not tested in ie.

**Comment by tdumitrescu** [comment]
Thanks for bringing this up. To help us reproduce, is there any way you could let us know:
- the full URL you're seeing this behavior on (subdomains are important)
- any custom configuration you're passing to `mixpanel.init()`

?

If any of this info is private or sensitive, just email support@mixpanel.com with it.

**Comment by vcarel** [comment]
Hi,
I'm on Chrome and the reset() method does not delete the mp_xxxx cookie on my side... Is it supposed to work ?

**Comment by vcarel** [comment]
It looks like reset takes time... but when I logout I do a page refresh so to ensure I get started with a new state. So it actually works from time to time.

Why not returning a Promise on reset ?

**Comment by tdumitrescu** [comment]
There's nothing about the `reset` code that ought to be anything but instantaneous (https://github.com/mixpanel/mixpanel-js/blob/f21586cf441b5461919bdba1d6257a2e9c084155/src/mixpanel-core.js#L1348-L1356 and https://github.com/mixpanel/mixpanel-js/blob/f21586cf441b5461919bdba1d6257a2e9c084155/src/mixpanel-core.js#L402-L413). This might be more quickly resolved for you by running through your particular setup with our quite knowledgeable and helpful support staff at support@mixpanel.com.

**Comment by vcarel** [comment]
So I'm getting a random behaviour. Sometimes it works, most of the time it
doesn't. There's no log or anything which can help me to debug.

Le mer. 22 févr. 2017 18:31, tdumitrescu <notifications@github.com> a
écrit :

**Comment by tdumitrescu** [comment]
Yes, I really think this is a case where you should contact support@mixpanel.com to debug 1-on-1 with someone.
